src/components/Modal/Modal.tsx

```diff
@@ -1,34 +1,141 @@
+import {
+  chakra,
+  Modal as ChakraModal,
+  ModalOverlay,
+  ModalContent,
+  ModalHeader,
+  ModalFooter,
+  ModalBody,
+  ModalCloseButton,
+  useDisclosure,
+  ButtonGroup,
+} from "@chakra-ui/react";
 import * as React from "react";
 
-export interface ModalProps {
-  /** ClassName that appears in addition to "modal" */
-  className?: string;
+import Button from "../Button/Button";
+import useWindowSize from "../../hooks/useWindowSize";
+
+interface BaseModalProps {
+  bodyContent?: string | JSX.Element;
+  closeButtonLabel?: string;
+  headingText?: string | JSX.Element;
   /** ID that other components can cross reference for accessibility purposes */
   id?: string;
+  isOpen?: boolean;
+  onClose?: () => void;
 }
 
-/** Full-screen modal that appears on top of the body of the page. */
-export default class Modal extends React.Component<ModalProps, any> {
-  componentDidMount() {
-    document.body.classList.add("no-scroll");
-  }
-
-  componentWillUnmount() {
-    document.body.classList.remove("no-scroll");
-  }
+export interface ModalProps {
+  buttonText?: string;
+  /** ID that other components can cross reference for accessibility purposes */
+  id?: string;
+  modalProps: BaseModalProps;
+}
 
-  render() {
-    const { id, className } = this.props;
+const BaseModal = chakra(
+  ({
+    bodyContent,
+    closeButtonLabel = "Close",
+    headingText,
+    id,
+    isOpen,
+    onClose,
+    ...rest
+  }: React.PropsWithChildren<BaseModalProps>) => {
+    const defaultSize = "xl";
+    const fullSize = "full";
+    const [size, setSize] = React.useState<string>(defaultSize);
+    const windowDimensions = useWindowSize();
+    React.useEffect(() => {
+      if (windowDimensions.width <= 320) {
+        setSize(fullSize);
+      } else {
+        setSize(defaultSize);
+      }
+    }, [windowDimensions.width]);
 
     return (
-      <div
-        // eslint-disable-next-line jsx-a11y/no-noninteractive-tabindex
-        tabIndex={0}
-        className={`modal ${className}`}
+      <ChakraModal
```

Comment:
This functionality needs to be implemented:
https://v0.chakra-ui.com/modal#modal-overflow-behavior

To do this, add the following prop to the modal component:
```
scrollBehavior=“inside"
```